core/api/src/app/support/chat.ts

```diff
@@ -0,0 +1,65 @@
+import { RepositoryError } from "@/domain/errors"
+import { AccountsRepository, UsersRepository } from "@/services/mongoose"
+import { Assistant } from "@/services/openai"
+
+export const getSupportChatMessages = async (accountId: AccountId) => {
+  const account = await AccountsRepository().findById(accountId)
```

Comment:
don't think account should be the entry point for this module.

Would expect something like:
```
const supportContext = await SupportContextRepository().findForAccountId(accountId)
```

Reply:
so you would expect another mongodb collection that would have the account -> "threadId" (renamed to supportChatId) relatinoship? even if this all there is to keep right now for now?

Reply:
In our code base we always run into adding things onto `Account`. It's a common (anti)-pattern to have some kind of god entity that is the base for looking up everything related. But it doesn't have to be that way - we can flip the dependency around.

For example consider wanting to find the user for an account (hypothetical example):
```
const account = AccountsRepository.findById(account)
const user = UserRepository.findById(account.ownerUserId)
```

In this case the dependency graph goes `Account -> User` (via the field `account.ownerUserId`).

But it is easy to avoid the 'bloat' on account by flipping the dependency graph:

```
const user = UserRepository.findByAccountId(accountId)
```

now `User -> Account` (via the field `user.accountId`).

So that is what I am suggesting here... not have `Account` know about `threadId` at all.
Rather have a `Thread` or `SupportChat` or something like that `Entity` that has an `id` (which currently is your `account.threadId`. And to lookup the entity have a `SupportChatRepository.findByAccountId(accountId)` query.

Reply:
.. so yes have a new mongodb table (to persist the entity).

Reply:
> Also I'm expecting this module to have its own entity instead of adding the reference to account.

ok I think this is my answer